src/client.toit

```diff
@@ -440,11 +440,16 @@ class Client:
       --content_type/string?
       --follow_redirects/bool:
 
-    if content_type and headers.get "Content-Type": throw "INVALID_ARGUMENT"
-    if headers.get "Transfer-Encoding": throw "INVALID_ARGUMENT"
-    if headers.get "Host": throw "INVALID_ARGUMENT"
-
-    if content_type:
+    has_content_header := false
+    if content_type and headers.single "Content-Type":
+      content_type_entries := headers.get "Content-Type"
+      if content_type_entries.size != 1 or content_type_entries[0] != content_type:
+        throw "INVALID_ARGUMENT"
+      has_content_header = true
+    if headers.single "Transfer-Encoding": throw "INVALID_ARGUMENT"
+    if headers.single "Host": throw "INVALID_ARGUMENT"
+
+    if content_type and not has_content_header:
```

Comment:
This seems overly complicated.

I see no reason to guard against there already being multiple
content-type headers.  If that's the case something already went
wrong earlier, and we would not discover it unless yet another
content-type was given.

```suggestion
    existing_content_type := headers.single "Content-Type"
    if not existing_content_type and existing_content_type != content_type: throw "INVALID_ARGUMENT"
    if headers.single "Transfer-Encoding": throw "INVALID_ARGUMENT"
    if headers.single "Host": throw "INVALID_ARGUMENT"

    if content_type:
```

We might want to use string.to_ascii_lower on the content-types before comparing them.

Reply:
refactored it, and used `to_ascii_lower`.